kratos/operations/operation.h

```diff
@@ -0,0 +1,191 @@
+    void operator()()
+    {
+        Execute();
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief This method creates an pointer of the process
+     * @details We consider as input a Mmodel and a set of Parameters for the sake of generality
+     * @warning Must be overrided in each process implementation
+     * @param rModel The model to be consider
+     * @param ThisParameters The configuration parameters
+     */
+    virtual Operation::Pointer Create(
+        Model& rModel,
+        Parameters ThisParameters) const
+    {
+        KRATOS_ERROR << "Calling base class Create. Please override this method in the corresonding Operation" << std::endl;
+        return nullptr;
+    }
+
+    /**
+     * @brief Execute method is used to execute the Operation algorithms.
+     */
+    virtual void Execute()
+    {
+        KRATOS_ERROR << "Calling base class Execute. Please override this method in the corresonding Operation" << std::endl;
+    }
+
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     */
+    virtual const Parameters GetDefaultParameters() const
+    {
+        KRATOS_ERROR << "Calling the base Operation class GetDefaultParameters. Please implement the GetDefaultParameters in your derived process class." << std::endl;
+        const Parameters default_parameters = Parameters(R"({})" );
+
+        return default_parameters;
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const
+    {
+        return "Operation";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const
+    {
+        rOStream << "Operation";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const
+    {
+    }
+
+    ///@}
+}; // Class Operation
+
+///@name Type Definitions
+///@{
+
+#ifndef KRATOS_REGISTER_OPERATION
+#define KRATOS_REGISTER_OPERATION(                                                                              \
+    module_name,                                                                                                \
+    operation_name,                                                                                             \
+    operation_prototype)                                                                                        \
+    {                                                                                                           \
+        std::string all_path = std::string("Operations.All.") + operation_name;                                 \
+        if (!Registry::HasItem(all_path)) {                                                                     \
```

Comment:
I understand that you are checking to send a custom error. Isn't it?

If this is the case, we may need to create (not in this PR) a AddItem(path, error_lambda) to make this more straightforward

Reply:
True. As it is right now we're checking twice, and the custom error isn't that custom... About the the `AddItem` with error thrown, it is not a bad idea.

Considering that we're about to start the PR with all the new registry stuff, and that this macro is almost identical to that of the `Process` that I'll add in there, I think we'd better discuss it in the upcoming one.

Reply:
Agree that we shall do it in a separate PR